src/scf/self_consistent_field.jl

```diff
@@ -181,42 +170,49 @@ Overview of parameters:
         Δρ = ρout - ρin
 
         # Update info with results gathered so far
-        info = (; ham, basis, converged, stage=:iterate, algorithm="SCF",
-                ρin, ρout, α=damping, n_iter, nbandsalg.occupation_threshold,
-                runtime_ns=time_ns() - start_ns, nextstate...,
-                diagonalization=[nextstate.diagonalization])
+        info_next = (; ham, basis, converged, stage=:iterate, algorithm="SCF",
+                       ρin, α=damping, n_iter, nbandsalg.occupation_threshold,
+                       runtime_ns=time_ns() - start_ns, nextstate...,
+                       diagonalization=[nextstate.diagonalization])
 
         # Compute the energy of the new state
         if compute_consistent_energies
             (; energies) = energy(basis, ψ, occupation; ρ=ρout, eigenvalues, εF)
         end
+        history_Etot = vcat(info.history_Etot, energies.total)
+        history_Δρ = vcat(info.history_Δρ, norm(Δρ) * sqrt(basis.dvol))
+        info_next = merge(info_next, (; energies, history_Etot, history_Δρ))
 
-        # Push energy and density change of this step.
-        push!(history_Etot, energies.total)
-        push!(history_Δρ,   norm(Δρ) * sqrt(basis.dvol))
-        info = merge(info, (; energies, history_Δρ, history_Etot, converged))
+        # Apply mixing and pass it the full info as kwargs
+        ρnext = ρin .+ T(damping) .* mix_density(mixing, basis, Δρ; info_next...)
 
-        converged = is_converged(info)
-        converged = MPI.bcast(converged, 0, MPI.COMM_WORLD)  # Ensure same converged
-        callback(merge(info, (; converged)))
+        callback(info_next)
+        if is_converged(info_next)
+            info_next = merge(info_next, (; converged=true))
+        end
```

Comment:
Be careful here, this is not doing the same thing (order of callback and is_converged swapped) and also you removed the MPI broadcast (which could explain the failing MPI tests)

Reply:
Thanks! Added back the MPI bcast, indeed MPI tests are passing now.